pinot-core/src/main/java/org/apache/pinot/core/operator/docidsets/SVScanDocIdSet.java

```diff
@@ -28,8 +28,12 @@ public final class SVScanDocIdSet implements FilterBlockDocIdSet {
 
   public SVScanDocIdSet(PredicateEvaluator predicateEvaluator, DataSource dataSource, int numDocs,
       boolean nullHandlingEnabled) {
-    _docIdIterator = new SVScanDocIdIterator(predicateEvaluator, dataSource.getForwardIndex(), numDocs,
-        dataSource.getNullValueVector(), nullHandlingEnabled);
+    if (nullHandlingEnabled) {
+      _docIdIterator = new SVScanDocIdIterator(predicateEvaluator, dataSource.getForwardIndex(), numDocs,
+          dataSource.getNullValueVector());
+    } else {
+      _docIdIterator = new SVScanDocIdIterator(predicateEvaluator, dataSource.getForwardIndex(), numDocs);
+    }
```

Comment:
Suggest changing it to
```suggestion
    NullValueVectorReader nullValueVector = nullHandlingEnabled ? dataSource.getNullValueVector() : null;
    _docIdIterator = new SVScanDocIdIterator(predicateEvaluator, dataSource.getForwardIndex(), numDocs, nullValueVector);
```